### PR TITLE
Add GitHub action for code scanner

### PR DESCRIPTION
**Pull request by 610v4nn1**
Add GitHub action for code scanner

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  schedule:
+    - cron: '28 5 * * 4'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
```
Comment (wistuba) [inline .github/workflows/codeql.yml:18]: only required for private repos

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'python' ]
```
Comment (wistuba) [inline .github/workflows/codeql.yml:24]: we can drop this

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main", dev ]
```
Comment (wistuba) [inline .github/workflows/codeql.yml:5]: why the "?

File: .github/workflows/codeql.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      with:
+        languages: ${{ matrix.language }}
+
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v2
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v2
+      with:
+        category: "/language:${{matrix.language}}"
```
Comment (wistuba) [inline .github/workflows/codeql.yml:42]: I don't think we need this

File: .github/workflows/codeql.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
```
Comment (wistuba) [inline .github/workflows/codeql.yml:34]: python

File: .github/workflows/codeql.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v2
```
Comment (wistuba) [inline .github/workflows/codeql.yml:37]: do we need this for python?

**Review by wistuba — CHANGES_REQUESTED** [review]
all knowledge taken from https://github.com/github/codeql-action

File: .github/workflows/codeql.yml (hunk 4 above)
Comment (lballes) [reply .github/workflows/codeql.yml:37]: What about this? Do we need it?

File: .github/workflows/codeql.yml (hunk 5 above)
Comment (lballes) [reply .github/workflows/codeql.yml:32]: Why not just python?

**Review by lballes — APPROVED** [review]

**Pull request merged**